serve/mlc_serve/model/model_common.py

```diff
@@ -36,6 +38,59 @@ def get_num_cache_blocks(
     )
 
 
+def get_raw_logprob_info(
+    logits,
+    token,
+    top_logprobs_num,
+) -> RawLogprobsInfo:
+    logprobs = torch.log_softmax(logits, dim=-1)
+    res_logprob = logprobs[token].cpu().numpy()
+
+    if top_logprobs_num == 0:
+        top_logprobs = None
+        top_tokens = None
+    else:
+        assert top_logprobs_num <= LOGPROB_TOP_K_MAX, "Invalid input top_logprobs"
+        top_logprobs, top_tokens = torch.topk(
+            logprobs, k=top_logprobs_num, dim=-1, largest=True, sorted=True
+        )
+        top_tokens=top_tokens.cpu().numpy()
+        top_logprobs=top_logprobs.cpu().numpy()
+
+    # Set to raw logprob info
+    return RawLogprobsInfo(
+        # TODO(vvchernov): it is number, cpu().numpy()?
+        current_token=token.cpu().numpy(),
```

Comment:
I think `.cpu().numpy()` can be removed here. `current_token` seems to be an element from `res_random` or `res_greedy`, which is already on cpu. Remember, `.cpu()` and `.numpy()` involve cudaMemcpy and another copy on CPU.